src/script/lua_api/l_env.cpp

```diff
@@ -1291,22 +1291,32 @@ int ModApiEnv::l_find_path(lua_State *L)
 	return 0;
 }
 
-// spawn_tree(pos, treedef)
+// spawn_tree(pos, treedef, [vmanip])
 int ModApiEnv::l_spawn_tree(lua_State *L)
 {
-	GET_ENV_PTR;
-
 	v3s16 p0 = read_v3s16(L, 1);
 
 	treegen::TreeDef tree_def;
-	const NodeDefManager *ndef = env->getGameDef()->ndef();
+	treegen::error e;
+	if (lua_gettop(L) < 3) {
```

Comment:
I^d rather do this with a `lua_isnoneornil(L, 3)` check so that modders could still pass `nil` in wrapper functions (easier to work with).

Reply:
Makes sense. I changed it.